### PR TITLE
TIP-808: Use version strategy for frontend assets

### PR DESCRIPTION
**Pull request by tamarasaurus**
**Description (for Contributor and Core Developer)**

This PR adds a version strategy for our JS and CSS assets so that we can bust the cache when we release a patch version. 

E.g. 
`<script type="text/javascript" src="/dist/manifest.min.js"></script>` becomes
`<script type="text/javascript" src="/dist/manifest.min.js?2.0.1 "></script>`

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/VersionStrategy/CacheBusterVersionStrategy.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    public function applyVersion($path)
+    {
+        $version = $this->getVersion($path);
+
+        if ('' === $version) {
+            return $path;
+        }
+
+        $versioned = sprintf('%s?%s', ltrim($path, '/'), $version);
```
Comment (nidup) [inline src/Pim/Bundle/EnrichBundle/VersionStrategy/CacheBusterVersionStrategy.php]: we could use php constant for the "/"?

File: src/Pim/Bundle/EnrichBundle/VersionStrategy/CacheBusterVersionStrategy.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Pim\Bundle\EnrichBundle\VersionStrategy;
+
+use Pim\Bundle\CatalogBundle\VersionProviderInterface;
+use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+
+class CacheBusterVersionStrategy implements VersionStrategyInterface
```
Comment (nidup) [inline src/Pim/Bundle/EnrichBundle/VersionStrategy/CacheBusterVersionStrategy.php]: could you add a spec test to help to figure out the exact behavior?

**Comment by nidup** [comment]
@BitOne interesting PR if you want take a look and give your opinion, thx :)

File: src/Pim/Bundle/EnrichBundle/VersionStrategy/CacheBusterVersionStrategy.php (hunk 1 above)
Comment (tamarasaurus) [reply src/Pim/Bundle/EnrichBundle/VersionStrategy/CacheBusterVersionStrategy.php]: LIke DIRECTORY_SEPARATOR ?

File: src/Pim/Bundle/EnrichBundle/VersionStrategy/CacheBusterVersionStrategy.php (hunk 2 above)
Comment (tamarasaurus) [reply src/Pim/Bundle/EnrichBundle/VersionStrategy/CacheBusterVersionStrategy.php]: Will do 👍

**Pull request merged**